diff --git a/devel_generate/src/Plugin/DevelGenerate/MenuDevelGenerate.php b/devel_generate/src/Plugin/DevelGenerate/MenuDevelGenerate.php
index 73582a7..d1ec7f8 100644
--- a/devel_generate/src/Plugin/DevelGenerate/MenuDevelGenerate.php
+++ b/devel_generate/src/Plugin/DevelGenerate/MenuDevelGenerate.php
@@ -326,6 +326,8 @@ class MenuDevelGenerate extends DevelGenerateBase implements ContainerFactoryPlu
 
     $nids = array();
     for ($i = 1; $i <= $num_links; $i++) {
+      $link_value = ['uri' => 'internal:/<front>', 'options' => ['devel' => TRUE]];
+
       // Pick a random menu.
       $menu_name = $menus[array_rand($menus)];
       // Build up our link.
@@ -337,7 +339,6 @@ class MenuDevelGenerate extends DevelGenerateBase implements ContainerFactoryPlu
         'bundle' => 'menu_link_content',
         'description' => $this->t('Description of @title.', array('@title' => $link_title)),
       ));
-      $link->link->options = array('devel' => TRUE);
 
       // For the first $max_width items, make first level links.
       if ($i <= $max_width) {
@@ -376,24 +377,17 @@ class MenuDevelGenerate extends DevelGenerateBase implements ContainerFactoryPlu
           $node = $select->execute()->fetchAssoc();
           if (isset($node['nid'])) {
             $nids[$menu_name][] = $node['nid'];
-            $link->link->uri = 'entity:node/' . $node['nid'];
+            $link_value['uri'] = 'entity:node/' . $node['nid'];
             $link->title = $node['title'];
             break;
           }
 
         case 'external':
-          $link->link->uri = 'http://www.example.com/';
-          break;
-
-        case 'front':
-          $link->link->uri = 'internal:/<front>';
-          break;
-
-        default:
-          $link->devel_link_type = $link_type;
+          $link_value['uri'] = 'http://www.example.com/';
           break;
       }
 
+      $link->link = $link_value;
       $link->save();
 
       $links[$link->id()] = $link->link_title;
