--- public_html/.htaccess	2019-12-04 08:44:18.000000000 +0000
+++ public_html/.htaccess	2019-12-12 09:29:25.749225500 +0000
@@ -44,8 +44,8 @@
   # Enable expirations.
   ExpiresActive On

-  # Cache all files for 2 weeks after access (A).
-  ExpiresDefault A1209600
+  # Cache all files for 1 year after access (A).
+  ExpiresDefault A31536000

   <FilesMatch \.php$>
     # Do not allow PHP scripts to be cached unless they explicitly send cache
@@ -76,6 +76,11 @@
   RewriteCond %{HTTPS} on
   RewriteRule ^ - [E=protossl:s]

+  # Not local development, force HTTPS
+  RewriteCond %{HTTP_HOST} !^(.+\.local|localhost|127\.0\.0\.1)$
+  RewriteCond %{HTTPS} off
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}/$1 [L,R=301]
+
   # Make sure Authorization HTTP header is available to PHP
   # even when running as CGI or FastCGI.
   RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
@@ -186,4 +191,22 @@
   Header always set X-Content-Type-Options nosniff
   # Disable Proxy header, since it's an attack vector.
   RequestHeader unset Proxy
+  # Always force HTTPS.
+  Header set Strict-Transport-Security "max-age=31536000" env=HTTPS
+</IfModule>
+
+<IfModule LiteSpeed>
+  # Cache static files for 1 year.
+  <filesMatch ".(jpg|jpeg|png|gif|ico|svg|css|js|gz|woff|woff2|ttf|eot)$">
+    Header set Cache-Control "public, max-age=31536000"
+    Header unset Expires
+    Header unset Last-Modified
+  </filesMatch>
+</IfModule>
+
+# php -- BEGIN cPanel-generated handler, do not edit
+# Set the “ea-php73” package as the default “PHP” programming language.
+<IfModule mime_module>
+  AddHandler application/x-httpd-ea-php73 .php .php7 .phtml
 </IfModule>
+# php -- END cPanel-generated handler, do not edit
